python/api-examples-source/tutorials/execution-flow/fragments/tutorial-fragment-rerun.py

```diff
@@ -0,0 +1,87 @@
+import streamlit as st
+import pandas as pd
+import numpy as np
+from datetime import date, timedelta
+import string
+import time
+
+
+@st.cache_data
+def get_data():
+    """Generate random sales data for Widget A through Widget Z"""
+
+    product_names = ["Widget " + letter for letter in string.ascii_uppercase]
+    average_daily_sales = np.random.normal(1_000, 300, len(product_names))
+    products = dict(zip(product_names, average_daily_sales))
+
+    data = pd.DataFrame({})
+    sales_dates = np.arange(date(2023, 1, 1), date(2024, 1, 1), timedelta(days=1))
+    for product, sales in products.items():
+        data[product] = np.random.normal(sales, 300, len(sales_dates)).round(2)
+    data.index = sales_dates
+    data.index = data.index.date
```

Comment:
Sorry I know it's a super nit and I'm bad at pandas, is this functionally different from just `data.index = sales_dates.date`? I saw the comment about stripping the timestamp but I was still left slightly confused about the double assignment.

Reply:
It will error if I try to use `.date` on the `numpy` object. Even though its defined with `date` instead of `dateime`, `pandas` insists on adding the zero timestamps, so (as far as I know) I have to clean it up after creating the `pandas` index object, but there may very well be something cleaner. I figured the `.date` on the index was a nice enough hint that would be more broadly applicable anyway.